### ISSUE TITLE
Thumbnail ratings

### ISSUE THREAD

**Issue by jonandeva** [YePpHa/YouTubeCenter#1814]
I have been trying to figure this out for probably 30 minutes. As has been noted previously, the ratings are not showing up on the video thumbnails anymore. I understand that the solution is to enable V3 API in "General Settings". I am using the latest version of YouTubeCenter on Chrome via tampermonkey an I cannot find this V3 option anywhere! Please help.

**Comment by Zarggg** [comment]
Please install the current [developer version](https://github.com/YePpHa/YouTubeCenter/wiki/Developer-Version) to ensure you are actually running the latest version of YTC.

**Comment by jonandeva** [comment]
Wow, that did it. I didn't even know that existed. Thanks.

**Comment by SINE** [comment]
Simply activating v3 API doesn't fix it, because the problem is caused due to the API-Key that YouTubeCenter is using being a free one and it's completely overloaded (too many YouTubeCenter users).

A little info from the YouTubeCenter main developer: https://github.com/YePpHa/YouTubeCenter/issues/1783

You can fix it by additionally using your own API Key. 
In order to get one, click on the little question mark next to the "API Key" field in the YouTubeCenter settings and then follow all steps (you need a Google Account for that).

After you get the key, copy-paste that into the settings field for the API Key.

Related issue:
#1817

**Comment by Long-Horse** [comment]
@SINE My ratings bar works after getting an API key and enabling it but commenter country flags still show a ? in it instead of the actual flag.

**Comment by SINE** [comment]
@Long-Horse try re-installing the script and resetting all settings, before applying the key.

**Comment by Long-Horse** [comment]
I tried reinstalling GreaseMonkey and the script and it still doesn't work.

**Comment by SINE** [comment]
@Long-Horse Do not re-install GreaseMonkey, re-Install the YouTubeCenter Script!.


First, go to YouTube and open the YouTubeCenter Menu.
On the bottom of the *General Settings* tab, select **Reset Settings**

To uninstall the script:

You can find that by typing **about:addons** in the address-bar (awesomebar, URL-bar, whatever you wanna call it),
on the left side choose "Userscripts".
Next to **YouTubeCenter Developer Build ...** select "remove".

Restart Firefox.

Install the Userscript by clicking [this link](https://github.com/YePpHa/YouTubeCenter/raw/master/dist/YouTubeCenter.user.js)

**Comment by Long-Horse** [comment]
Sorry I meant TamperMonkey. I did this and they still don't work. Am I able to restore my old settings with my .ytcs file? I would rather keep my old settings than have to re-do them all. I've changed quite a bit from default.

**Comment by SINE** [comment]
Can't help you directly with Chrome.
You can restore the settings with your settings-file, yes.

**Comment by ireun** [comment]
@SINE, @Long-Horse  Yup, YT API doesn't contain country ifo now, it has been discussed in #1877. 

*Closing..*

**Comment by Fironet** [comment]
After pasting in the v3 API, and downloading and installing the dev build thumbnail ratings are still not being displayed on Youtube thumbnails.

I feel like this issue should be open.

**Comment by SINE** [comment]
You should read the post above you. There is no way to return a function that has been removed from google. For that, you'd have to make your own service which would crawl every youtube video and update the ratings for you - OR - synchronize the ratings by parsing them from other script users, but that would also have to go through some kind of database server.

The usefullness is in no relation to the work behind such a concept.

So: No return for the function - Issue closed

**Comment by Fironet** [comment]
> You should read the post above you. There is no way to return a function that has been removed from Google.

Which post mentioned this? If it has been removed from the API that's understandable but didn't see that discussed above.
